pkgs/development/python-modules/gpt4all-bindings/default.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  gpt4all,
+  gpt4all-cuda,
+  buildPythonPackage,
+  stdenv,
+  requests,
+  tqdm,
+  pythonOlder,
+  setuptools,
+  wheel,
+  python,
+  cuda ? false,
+}:
+let
+  basepkg = if cuda then gpt4all-cuda else gpt4all;
+
+  gpt4all-backend = stdenv.mkDerivation {
+    pname = "gpt4all-backend";
+    inherit (basepkg)
+      src
+      version
+      cmakeFlags
+      nativeBuildInputs
+      buildInputs
+      ;
+
+    sourceRoot = "${basepkg.src.name}/gpt4all-backend";
+    installPhase = ''
+      cp -r ./ $out
+    '';
+  };
+in
+buildPythonPackage {
+  pname = "gpt4all-bindings";
+  inherit (basepkg) src version;
+  sourceRoot = "${basepkg.src.name}/gpt4all-bindings/python";
+  pyproject = false;
```

```suggestion
  pyproject = true;
```
A typical python package with setup.py.

https://github.com/nomic-ai/gpt4all/tree/main/gpt4all-bindings/python